### PR TITLE
[Synapse Platform Service - Synase Link] add GA version 

### LINKED ISSUE #21728: [Synapse Platform Service - Synase Link] API Review
New API Review meeting has been requested.

**Service Name**: Azure Synapse Platform Service - Synase Link
**Review Created By**: Zhicheng Yang @ZhichengYang-MS 
**Review Date**: 12/8/2022 4:00 PM PT
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/21665
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: Add GA version for Synapse Link

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=7d0da6c3-e771-ed11-9562-000d3a3418aa)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by ZhichengYang-MS**
# Data Plane API - Pull Request
add GA version of Synapse Link, see [Synapse Link for SQL](https://learn.microsoft.com/en-us/azure/synapse-analytics/synapse-link/sql-synapse-link-overview)

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [x] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

fix https://github.com/Azure/azure-rest-api-specs/issues/21728

### CODE REVIEW

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json [hunk 1]
```diff
@@ -0,0 +1,1011 @@
...
+        "parameters": [
+          {
+            "$ref": "#/parameters/LinkConnectionName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (DominikMe) [inline specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json:158]: Why do we have a 200 response in addition to the 204?

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json [hunk 2]
```diff
@@ -0,0 +1,1011 @@
...
+      "description": "The structure options of a target table",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Target table structure type"
+        }
+      }
+    },
+    "LinkConnectionDetailedStatus": {
+      "type": "object",
```
Comment (DominikMe) [inline specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json:815]: Are any of the properties supposed to be required?

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json [hunk 3]
```diff
@@ -0,0 +1,1011 @@
...
+      }
+    },
+    "LinkConnectionRefreshStatus": {
+      "type": "object",
+      "description": "The link connection refresh status",
+      "properties": {
+        "refreshStatus": {
+          "type": "string",
+          "description": "Link connection refresh status",
+          "readOnly": true
```
Comment (mikekistler) [inline specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json:973]: Don't mark properties as `readOnly` in schemas that are never part of a request body.

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json [hunk 4]
```diff
@@ -0,0 +1,1011 @@
...
+        },
+        "errorMessage": {
+          "type": "string",
+          "description": "Link connection refresh error message",
+          "readOnly": true
```
Comment (mikekistler) [inline specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json:978]: Don't mark properties as `readOnly` in schemas that are never part of a request body.

**Review by mikekistler — APPROVED** [review]
Looks good. 👍 

I flagged a few minor issues that should be easy to fix.

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json (hunk 2 above)
Comment (ZhichengYang-MS) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json:815]: LinkConnectionDetailedStatus is required by "/linkconnections/{linkConnectionName}/detailedstatus"

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json [hunk 5]
```diff
@@ -0,0 +1,1011 @@
...
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/linkconnections/{linkConnectionName}": {
+      "put": {
+        "tags": [
+          "linkConnections"
+        ],
+        "operationId": "LinkConnection_CreateOrUpdate",
```
[the remark this answers is not in the document]
Comment (ZhichengYang-MS) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json:70]: We`d like to be consistent in wording as previous versions. Could we keep that?

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json (hunk 3 above)
Comment (ZhichengYang-MS) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json]: removed

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json (hunk 4 above)
Comment (ZhichengYang-MS) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json]: removed

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json (hunk 1 above)
Comment (ZhichengYang-MS) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json:158]: When deleting LinkConnection, our service will call our dependent metadata service, which stores the LinkConnection info and just returns the response of it. And dependent metadata service changed behavior (200/204) of deleting metadata. So, to be compatible with two kinds of behavior, we would like to keep both 200 and 204.

File: specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json [hunk 6]
```diff
@@ -0,0 +1,1011 @@
...
+            "$ref": "#/parameters/LinkConnectionName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
```
[the remark this answers is not in the document]
Comment (ZhichengYang-MS) [reply specification/synapse/data-plane/Microsoft.Synapse/stable/2023-01-01/linkConnections.json:160]: When deleting LinkConnection, our service will call our dependent metadata service, which stores the LinkConnection info and just returns the response of it. And dependent metadata service changed behavior (200/204) of deleting metadata. So, to be compatible with two kinds of behavior, we would like to keep both 200 and 204.

**Comment by ZhichengYang-MS** [comment]
Hi, @mikekistler, @weidongxu-microsoft, I have fixed or replied to the comments. May I know how could I proceed?

**Comment by ZhichengYang-MS** [comment]
Hi, @mikekistler, @weidongxu-microsoft, May I know how could I proceed?

**Review by mikekistler — CHANGES_REQUESTED** [review]
I think we need to confirm that the service verifies the api-version before we can approve this for GA.

**Pull request closed without merging**